src/plugins/auto/src/plugin.cpp

```diff
@@ -275,258 +317,243 @@ InferenceEngine::Parameter MultiDeviceInferencePlugin::GetConfig(const std::stri
     // set performanceHint for AutoExecutableNetwork
-    autoSContext->_performanceHint = loadConfig.get_property(ov::hint::performance_mode.name()).as<std::string>();
+    auto_s_context->m_performance_hint = load_config.get_property(ov::hint::performance_mode.name());
     // filter the device that supports filter configure
-    metaDevices = ParseMetaDevices(strDevices, fullConfig);
-    auto supportDevicesByConfig = FilterDevice(metaDevices, filterConfig);
-    if (supportDevicesByConfig.empty()) {
-        IE_THROW() << "There is no device support the configure";
-    }
-    auto supportDevices = supportDevicesByConfig;
-    CNNNetwork clonedNetwork;
-    std::string clonedModelPath = modelPath;
-    // reset the strDevices to support devices
-    strDevices = "";
-    // calling GetValidDevices() to get a prioritized list of devices
-    bool isCumulative =
-        (autoSContext->_performanceHint == IE::PluginConfigParams::CUMULATIVE_THROUGHPUT) ? true : false;
-    std::list<DeviceInformation> devicesWithPriority(supportDevices.begin(), supportDevices.end());
-    if (modelPath.empty()) {
-        // if network is valid
-        LOG_INFO_TAG("load with CNN network");
-        supportDevices = FilterDeviceByNetwork(supportDevicesByConfig, network);
-        clonedNetwork = InferenceEngine::details::cloneNetwork(network);
-        // clone the network, in case of reshape conflict
-    } else {
-        // model path, enable model load with single device situation
-        if (supportDevices.size() > 1 && !isCumulative) {
-            clonedNetwork = GetCore()->ReadNetwork(modelPath, std::string());
-            // do we really need to disable model path?
-            clonedModelPath = "";
-            LOG_INFO_TAG("load with CNN network");
-        } else {
-            LOG_INFO_TAG("load with model path");
+    meta_devices = parse_meta_devices(str_devices, full_property);
+    auto support_devices_by_property = filter_device(meta_devices, filter_property);
+    if (support_devices_by_property.empty()) {
+        OPENVINO_THROW("There is no device support the current configure");
+    }
+    auto support_devices = support_devices_by_property;
+    // reset the str_devices to support devices
+    str_devices = "";
+    bool is_cumulative =
+        (auto_s_context->m_performance_hint == ov::hint::PerformanceMode::CUMULATIVE_THROUGHPUT) ? true : false;
+    std::list<DeviceInformation> devices_with_priority(support_devices.begin(), support_devices.end());
+    std::shared_ptr<ov::Model> cloned_model, ppp_model;
+    if (model_path.empty()) {
+        support_devices = filter_device_by_network(support_devices_by_property, model);
+        cloned_model = model->clone();
+        ppp_model = cloned_model->clone();
+        ov::preprocess::PrePostProcessor preproc(ppp_model);
+        // temp solution to reolve the prceision/layout mismatch between new/old api
+        if (!is_new_api()) {
```

Comment:
yes, you are right.

Reply:
actually it should be "once we switched to 2.0 public API", this temp solution is because for compliance with old public preprocessing API, there is extra ppp added in CoreImpl for hardware plugin (but not for virtual plugin):
https://github.com/openvinotoolkit/openvino/blob/master/src/inference/src/dev/core_impl.cpp#L600

Reply:
should we also wrap it with deprecation suppression marcos?